src/main/java/argelbargel/jenkins/plugins/gitlab_branch_source/api/GitLabAPI.java

```diff
@@ -278,41 +279,35 @@ private boolean unregisterProjectHook(String url, int projectId) throws IOExcept
                 return true;
             }
         }
-
         return false;
     }
 
     private String projectUrl(String group, GitLabProjectSelector selector, GitLabProjectVisibility visibility, String searchPattern) {
         StringBuilder urlBuilder = new StringBuilder(GitlabGroup.URL).append(PATH_SEP).append(group).append(GitLabProject.URL).append("?membership=true");
-
         if (!VISIBLE.equals(selector)) {
             urlBuilder.append("&").append(selector.id()).append("=true");
         }
-
         if (!ALL.equals(visibility)) {
-            urlBuilder.append("&").append("visibility=").append(visibility.id());
+            urlBuilder.append("&visibility=").append(visibility.id());
         }
-
         if (!StringUtils.isEmpty(searchPattern)) {
-            urlBuilder.append("&").append("search=").append(searchPattern);
+            urlBuilder.append("&search=").append(searchPattern);
         }
-
         return urlBuilder.toString();
     }
 
 
     private String projectUrl(GitLabProjectSelector selector, GitLabProjectVisibility visibility, String searchPattern) {
-        StringBuilder urlBuilder = new StringBuilder(GitlabProject.URL)
-                .append(PATH_SEP).append(selector.id()).append("?membership=true");
-
+        StringBuilder urlBuilder = new StringBuilder(GitlabProject.URL).append("?membership=true");
+        if (!VISIBLE.equals(selector)) {
+            urlBuilder.append("&").append(selector.id()).append("=true");
+        }
```

Comment:
@imduffy15 I pulled your changes and I think the reason I didn't have the issue you experienced is that I'm filtering on a group and maybe hitting the above `projectUrl` method and not this one.   I'm assuming this block was in there for the same fix, so I replaced your two conditionals with this one for consistency.  Does this look right to you?